Introduce safe_delay()

AnHardt opened the pull request:
Introduce safe_delay() for long delays what otherwise would cause
watchdog resets. Especially when added.

Explizit `lcd_set_custom_characters(false)` at the end of `bootscreen()`. 
Else the charset is not proper set in case of `LCD_PROGRESS_BAR`.

Follow up for #4032

AnHardt commented:
Likely this is useful enough to give it a more prominent place. Marlin_main.cpp? temperature.cpp?

thinkyhead commented on Marlin/ultralcd_implementation_hitachi_HD44780.h:
```diff
@@ -443,6 +443,15 @@ unsigned lcd_print(char c) { return charset_mapper(c); }
     lcd.setCursor(indent, 2); lcd.print('\x02'); lcd_printPGM(PSTR( "------" ));  lcd.print('\x03');
   }
 
+  void save_delay(uint16_t del){
```
I think you mean "`safe_delay`."

ghost commented:
Thanks for the trying to fix.
Problem of LED flasing with `USE_WATCHDOG` has been solved.
But, problem of broken characters at first time is still alive...
About "FR", lower half part is replaced by "clock" ?

![dscn1632](https://cloud.githubusercontent.com/assets/15308844/16080263/ea6bab8c-3342-11e6-9a8b-ce1564bf1016.JPG)

 A branch that it was used for this test:
https://github.com/esenapaj/Marlin/tree/testes

AnHardt commented:
@esenapaj 
I hope i got it now.

AnHardt committed:
Introduce save_delay()

Introduce save_delay() for long delays what otherwise would cause
watchdog resets

Explizit lcd_set_custom_characters(false) at the end of bootscreen().
Else the charset is not proper set in case of LCD_PROGRESS_BAR.

ghost commented:
Thanks for the updating.
Now problem is almost solved, but still bit alive.
Character of thermistor is OK, but "°" is broken (not always. It's extremely rare. < 5% per reset?).
![dscn1634](https://cloud.githubusercontent.com/assets/15308844/16088821/c3a1de06-3363-11e6-8bdb-50c0f0da1ffc.JPG)

"FR" is often broken(70~80% per reset?).
![dscn1633](https://cloud.githubusercontent.com/assets/15308844/16088479/48ea83da-3362-11e6-8ddf-06d8eef6d3de.JPG)

AnHardt commented:
I can reproduce the problem occasionally. What we see is the upper half of the 'bed symbol' at the place the '°' should be.  I have no idea how that can be.

thinkyhead commented:
> What we see is the upper half of the 'bed symbol' at the place the '°' should be.

I have always seen custom character issues on the first boot. It fixes itself if you re-display the Info Screen. Maybe the custom character code is running early, when the display isn't ready.
